pkg/dependencies/factory.go

```diff
@@ -653,6 +653,19 @@ func (f *Factory) WithClusterManager(clusterConfig *v1alpha1.Cluster) *Factory {
 	return f
 }
 
+func (f *Factory) WithMaxWaitPerMachine(maxWaitPerMachine time.Duration) *Factory {
+	f.buildSteps = append(f.buildSteps, func(ctx context.Context) error {
+		if f.dependencies.ClusterManager != nil {
+			return nil
+		}
+
+		f.dependencies.ClusterManager.SetMaxWaitPerMachine(maxWaitPerMachine)
```

Comment:
Instead of doing this, which relays on order (if cluster manager is nil, this will fail), I would just store `CliConfig` in the factory

Then, when you build ClusterManager, pass the `MaxWaitPerMachine` to its constructor